transformer_lens/utils.py

```diff
@@ -886,3 +886,62 @@ def get_causal_mask_for_left_padding(
         )
 
     return mask
+
+
+def locally_override_and_restore_defaults(function):
+    """
+    This decorator is used to override the parameters with default values during a function's execution.
+    It guarantees that the default parameters are not changed after the function execution,
+    even when an error occurs during the execution.
+    """
+    def wrapper(self, *args, **kwargs):
+        sig = inspect.signature(function)
+        arg_names = list(sig.parameters.keys())
+        if arg_names[0] == "self":
+            arg_names = arg_names[1:]
+
+        arg_values = dict(zip(arg_names, args))
+        arg_values.update(kwargs)
+
+        # prepend_bos
```

Comment:
I assume this comment section should be uncommented or removed?

Reply:
Hi Jay, you are right! I was actually considering hiding a refactoring commit of the overriding logic of `prepend_bos`, which was the reason why I just commented it out temporarily, but it will be better to separate the PRs.